hedera-mirror-rest/service/blockService.js

```diff
@@ -0,0 +1,65 @@
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+'use strict';
+
+const BaseService = require('./baseService');
+const {RecordFile} = require('../model');
+
+/**
+ * BlockService retrieval business logic
+ */
+class BlockService extends BaseService {
+  constructor() {
+    super();
+  }
+
+  buildWhereSqlStatement(whereQuery) {
+    let where = 'where true=true';
```

Comment:
Would prefer a cleaner sql statement without `where true=true` when no conditions are applicable
Maybe something like this
```suggestion
    let where = '';
    const params = [];
    if (whereQuery.length > 0) {
      for (let i = 1; i <= whereQuery.length; i++) {
        where += ` and ${whereQuery[i - 1].query} $${i} `;
        params.push(whereQuery[i - 1].param);
      }
    }
```